app/controllers/sample_types_controller.rb

```diff
@@ -133,43 +145,46 @@ def filter_for_select
     render partial: 'sample_types/select/filtered_sample_types'
   end
 
-  def batch_upload
-
-  end
+  def batch_upload; end
 
   private
 
+  def update_sample_json_metadata
+    # UpdateSampleMetadataJob.new(@sample_type).queue_job
+    UpdateSampleMetadataJob.new(@sample_type).perform_now
```

Comment:
should be `UpdateSampleMetadataJob.perform_later(@sample_type)` and only triggered when neccessary

Reply:
since this is quite common across several controllers, I'm wondering if it should go in a module and include checks for whether sample type attributes have changed in a particular way